src/validation_workflow/validation.py

```diff
@@ -18,11 +19,18 @@ def __init__(self, args: ValidationArgs) -> None:
         super().__init__()
         self.args = args
 
+    def check_url(self, url: str) -> bool:
+        if self.is_url_valid(url) and self.download(url, self.tmp_dir):  # type: ignore
```

Comment:
Thanks for the review @dblock,
Yes,` is_url_valid` does a HEAD request, and `download` downloads a file in a sequential manner at this point.

Right now in the line `if self.is_url_valid(url) and self.download(url, self.tmp_dir):` since `and` operator short circuits the download method is not called, it is only called when the url is valid and the validation part completely relies on the downloaded artifact as the ultimate goal is to validate this artifact for a respective distribution.

I thinks it's good to have a double verification before we download the artifact, and I can call the two methods seperately inside the check_url method if I'm not wrong and return a different exception when failed in each case as you said , let me know
Thanks.

Reply:
Why do you need to double verify? In which case will download fail when HEAD succeeds?

Reply:
Makes sense @dblock , can I go by implementing only download(url, self.tmp_dir) and ignore is_url_valid(url). This gets the required job done.

Reply:
Th idea of validation workflow is to verify that the OS and OSD artifacts from the OpenSearch official site can be successfully downloaded and to validate them as in the [documentation](https://opensearch.org/docs/latest/install-and-configure/install-opensearch/tar/) , mainly used to validate the artifacts for a particular version during releases and is independent of distribution and architecture. The base for the mechanism are the files that are downloaded from the official site.

Reply:
I see. The method names and the location of downloading is confusing then. There should be a `download` step that is clearly doing that.